monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js

```diff
@@ -382,6 +390,52 @@ class ConfigurePageComponent extends AuthComponent {
       }));
   }
 
+  sendLinuxMasqueStrings() {
+    const linuxMasqueStrings = this.state.masqueStrings?.linux_masque_strings;
+    const linuxMasqueBytes = this.getNullTerminatedBytesFromStrings(linuxMasqueStrings);
+
+    return IslandHttpClient.put(
+      APIEndpoint.linuxMasque, linuxMasqueBytes, true, 'bytes')
+        .then(res => {
+        if (res.status !== 204) {
+          throw Error();
+        }
+        return res;
+      })
+      .catch((error) => {
+        console.log(`bad configuration ${error}`);
+        this.setState({lastAction: 'invalid_configuration'});
+      });
+  }
+
+  sendWindowsMasqueStrings() {
+    const windowsMasqueStrings = this.state.masqueStrings?.windows_masque_strings;
+    const windowsMasqueBytes = this.getNullTerminatedBytesFromStrings(windowsMasqueStrings);
+
+    return IslandHttpClient.put(
+      APIEndpoint.windowsMasque, windowsMasqueBytes, true, 'bytes')
+        .then(res => {
+        if (res.status !== 204) {
+          throw Error();
+        }
+        return res;
+      })
+      .catch((error) => {
+        console.log(`bad configuration ${error}`);
+        this.setState({lastAction: 'invalid_configuration'});
+      });
+  }
```

Comment:
Can we extract a function, pass in an API Endpiont and a specific variable from the state, and reduce duplication?